applications/GeoMechanicsApplication/custom_elements/U_Pw_small_strain_FIC_element.cpp

```diff
@@ -14,12 +14,22 @@
 // Application includes
 #include "custom_elements/U_Pw_small_strain_FIC_element.hpp"
 #include "custom_utilities/equation_of_motion_utilities.h"
+#include "custom_utilities/extrapolation_utilities.h"
 #include "custom_utilities/math_utilities.h"
 #include "custom_utilities/transport_equation_utilities.hpp"
 
 namespace Kratos
 {
 
+auto CalculateSquareExtrapolationMatrix = [](Element* pElement) {
+    const auto extrapolation_matrix = ExtrapolationUtilities::CalculateExtrapolationMatrix(
+        pElement->GetGeometry(), pElement->GetIntegrationMethod(), pElement->Id(), nullptr);
+    if (extrapolation_matrix.size1() != extrapolation_matrix.size2()) {
+        KRATOS_ERROR << "Extrapolation matrix is not square for element id " << pElement->Id() << std::endl;
+    }
+    return extrapolation_matrix;
+};
```

Comment:
I think it's clearer to most programmers to define a free function (in an unnamed namespace) rather than a lambda expression. Also, I think you can make the pointed-to element `const`:
```suggestion
namespace
{

using namespace Kratos;

auto CalculateSquareExtrapolationMatrix (const Element* pElement)
{
    const auto result = ExtrapolationUtilities::CalculateExtrapolationMatrix(
        pElement->GetGeometry(), pElement->GetIntegrationMethod(), pElement->Id(), nullptr);
    if (result.size1() != result.size2()) {
        KRATOS_ERROR << "Extrapolation matrix is not square for element id " << pElement->Id() << std::endl;
    }
    return result;
}

}
```

Reply:
changed